actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -231,12 +231,21 @@ def normalize_options!(options, path_params, modyoule)
               if to.nil?
                 controller = default_controller
                 action = default_action
-              elsif to.is_a?(String) && to.include?("#")
-                to_endpoint = to.split("#").map!(&:-@)
-                controller  = to_endpoint[0]
-                action      = to_endpoint[1]
               else
-                raise ArgumentError, ":to must respond to `action` or `call`, or it must be a String that includes '#'"
+                if to.is_a?(String)
```

Comment:
I feel like this new code is much harder to follow than before #50466.

What do you think of this diff? It also passes the reproduction script and I think is a bit simpler:
```diff
diff --git a/actionpack/lib/action_dispatch/routing/mapper.rb b/actionpack/lib/action_dispatch/routing/mapper.rb
index 724bf72e5c..b152de0ea7 100644
--- a/actionpack/lib/action_dispatch/routing/mapper.rb
+++ b/actionpack/lib/action_dispatch/routing/mapper.rb
@@ -231,10 +231,15 @@ def normalize_options!(options, path_params, modyoule)
               if to.nil?
                 controller = default_controller
                 action = default_action
-              elsif to.is_a?(String) && to.include?("#")
-                to_endpoint = to.split("#").map!(&:-@)
-                controller  = to_endpoint[0]
-                action      = to_endpoint[1]
+              elsif to.is_a?(String)
+                if to.include?("#")
+                  to_endpoint = to.split("#").map!(&:-@)
+                  controller  = to_endpoint[0]
+                  action      = to_endpoint[1]
+                else
+                  controller = default_controller
+                  action = default_action
+                end
               else
                 raise ArgumentError, ":to must respond to `action` or `call`, or it must be a String that includes '#'"
               end
```

Reply:
Yeah it's nicer. I had to use `to` instead of `default_action` in the new branch handling the string when it's missing the `#`, otherwise the repro script wasn't passing.